### PR TITLE
Configure version in one place.

### PR DESCRIPTION
**Pull request by schmmd**
Fixes https://github.com/allenai/allennlp/issues/614

### CODE REVIEW

**Comment by schmmd** [comment]
@joelgrus should these be all caps constants or should I organize things another way (class, etc)?  This works for the documentation and setup.py--although I will do some more testing.

**Comment by joelgrus** [comment]
pylint will be mad if you don't make them all caps

**Comment by joelgrus** [comment]
if it were me, I would probably do something like

```
_MAJOR = 0
_MINOR = 3
_REVISION = "1-unreleased"

VERSION_SHORT = f"{_MAJOR}.{_MINOR}"
VERSION = f"{_MAJOR}.{_MINOR}.{_REVISON}"
```

you can play with the names of the last two

File: allennlp/version.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+_MAJOR = "0"
+_MINOR = "3"
+_REVISION = "1-unreleased"
+
+VERSION_SHORT = "{0}.{1}".format(_MAJOR, _MINOR)
+VERSION = "{0}.{1}.{2}".format(_MAJOR, _MINOR, _REVISION)
```
Comment (matt-gardner) [inline allennlp/version.py]: Can you add `from allennlp.version import VERSION as __version__` to `allennlp.__init__.py`?  I think that's the only thing missing here.  No strong preference between `VERSION` and `__version__` in the `__init__.py`, it just seems like `__version__` is more common.  We could also just give both...

**Pull request merged**